x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/BlockBuilderCopyFromTests.java

```diff
@@ -89,7 +90,16 @@ private void assertEvens(Block block) {
         Block.Builder builder = elementType.newBlockBuilder(block.getPositionCount() / 2, blockFactory);
         List<List<Object>> expected = new ArrayList<>();
         for (int i = 0; i < block.getPositionCount(); i += 2) {
-            builder.copyFrom(block, i, i + 1);
+            switch (elementType) {
+                case BOOLEAN -> ((BooleanBlockBuilder) builder).copyFrom((BooleanBlock) block, i);
+                case BYTES_REF -> ((BytesRefBlockBuilder) builder).copyFrom((BytesRefBlock) block, i, new BytesRef());
+                case DOUBLE -> ((DoubleBlockBuilder) builder).copyFrom((DoubleBlock) block, i);
+                case FLOAT -> ((FloatBlockBuilder) builder).copyFrom((FloatBlock) block, i);
+                case INT -> ((IntBlockBuilder) builder).copyFrom((IntBlock) block, i);
+                case LONG -> ((LongBlockBuilder) builder).copyFrom((LongBlock) block, i);
+                default -> throw new IllegalArgumentException();
+            }
```

Comment:
I could intentionally didn't make a version of this thing in `Block.Builder` because I:
1. Want you to know if you need a scratch.
2. Want you to specialize on each type in the tight loops.